lib/matrix_client/poller.ex

```diff
@@ -75,18 +85,29 @@ defmodule M51.MatrixClient.Poller do
     # Need to be larger than the timeout above (both in milliseconds)
     options = [recv_timeout: 1_000_000]
 
+    delay =
+      if delay do
+        Process.sleep(delay)
+        Kernel.min(delay * @connect_delay_factor, @connect_delay_max)
+      else
+        @connect_delay_min
+      end
+
     case M51.Matrix.RawClient.get(raw_client, path, [], options) do
       {:ok, events} ->
         handle_events(sup_pid, is_backlog, events)
         events["next_batch"]
 
       {:error, code, _} when code >= 500 and code < 600 ->
-        # server error, try again
-        poll_one(sup_pid, since, raw_client)
+        # server request processing error, try again
+        poll_one(sup_pid, since, raw_client, delay)
 
-      {:error, nil, :closed} ->
-        # server closed connection, likely due to timeout, retry
-        poll_one(sup_pid, since, raw_client)
+      {:error, nil, reason} ->
+        if delay do
```

Comment:
I don't think `delay` can be `nil` here

Reply:
It will be on first (immediate) reconnection attempt, which is whole point of this check - to not log issues that can be fixed by just that, only ones that persist.

Reply:
Hm, no, you are right, it'll be indeed set to min by then, my bad, will fix.